checker/src/org/checkerframework/checker/regex/RegexQualifiedTransfer.java

```diff
@@ -45,82 +59,218 @@ public RegexQualifiedTransfer(QualAnalysis<QualParams<Regex>> analysis) {
-                // (No special case is needed for asRegex(String) because of
-                // the annotation on that method's definition.)
-
-                // add annotation with correct group count (if possible,
-                // regex annotation without count otherwise)
-                QualParams<Regex> regex;
-                Node count = n.getArgument(1);
-                if (count instanceof IntegerLiteralNode) {
-                    IntegerLiteralNode iln = (IntegerLiteralNode) count;
-                    Integer groupCount = iln.getValue();
-                    regex = new QualParams<>(new GroundQual<Regex>(new RegexVal(groupCount)));
-                } else {
-                    regex = new QualParams<>(new GroundQual<Regex>(new RegexVal(0)));
-                }
-                QualValue<QualParams<Regex>> newResultValue = analysis
-                        .createSingleAnnotationValue(regex,
-                                result.getResultValue().getType().getUnderlyingType().getOriginalType());
-                return new RegularTransferResult<>(newResultValue,
-                        result.getRegularStore());
+        if (receiver instanceof ClassNameNode) {
+            ClassNameNode cnn = (ClassNameNode) receiver;
+            String receiverName = cnn.getElement().toString();
+            if (isRegexUtil(receiverName)) {
+                result = handleRegexUtil(n, method, result);
             }
         }
-
         return result;
     }
 
+    private TransferResult<QualValue<QualParams<Regex>>, QualStore<QualParams<Regex>>> handleRegexUtil(
+            MethodInvocationNode n, ExecutableElement method,
+            TransferResult<QualValue<QualParams<Regex>>, QualStore<QualParams<Regex>>> resultIn) {
+        if (ElementUtils.matchesElement(method, IS_REGEX_METHOD_NAME, String.class, int.class)) {
+            // RegexUtil.isRegex(s, groups) method
+            // (No special case is needed for isRegex(String) because of
+            // the annotation on that method's definition.)
+
+            QualStore<QualParams<Regex>> thenStore = resultIn.getRegularStore();
+            QualStore<QualParams<Regex>> elseStore = thenStore.copy();
+            ConditionalTransferResult<QualValue<QualParams<Regex>>, QualStore<QualParams<Regex>>> newResult =
+                    new ConditionalTransferResult<>(resultIn.getResultValue(), thenStore, elseStore);
+            FlowExpressionContext context = FlowExpressionParseUtil.buildFlowExprContextForUse(n,
+                    analysis.getContext());
+            Receiver firstParam;
+            try {
+                // TODO: is this the easiest way to do this?
+                firstParam = FlowExpressionParseUtil.parse("#1", context,
+                        analysis.getContext().getTypeFactory().getPath(n.getTree()));
+            } catch (FlowExpressionParseException e) {
+                firstParam = null;
+                assert false;
```

Comment:
Many of your comments were to previously existing code, so I can't give the best motivation.
In this case, the FlowExpression is the constant "#1", so it can never fail to parse.
However, I added the TODO comment, because I found this a rather strange approach to access the first argument of a call.